extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/ContextualErrorMessagesSystemPropertyTest.java

```diff
@@ -0,0 +1,42 @@
+package io.quarkus.rest.client.reactive;
+
+import org.assertj.core.api.Assertions;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.jboss.resteasy.reactive.ClientWebApplicationException;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.rest.client.reactive.configuration.EchoResource;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class ContextualErrorMessagesSystemPropertyTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest TEST = new QuarkusUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(EchoResource.class, HelloClient2.class)
+                    .addAsResource(new StringAsset(
+                            "quarkus.rest-client.hello2.url=http://localhost:${quarkus.http.test-port:8081}/wrong-url"),
+                            "application.properties"));
+
+    static {
+        System.setProperty("quarkus.rest-client.disable-contextual-error-messages", "true");
```

Comment:
This is quite dangerous to be honest. Would it not work to have this in a `@BeforeAll` call and also unset or reset the property in an `@AfterAll`?

Reply:
I will look at that.

Reply:
Thanks

Reply:
Regarding "dangerous", do you mean the system property could leak into other tests or is it something else? Thanks :)

Reply:
`@BeforeAll` doesn't work, it looks like config root is initialized sooner still. I will try to think about it more tomorrow, in the worst case we can remove this one test...

Reply:
Yeah exactly, I'm worried about system property leaks